File: Java/util/InputValidator.java
package util;

import java.util.Scanner;

public class InputValidator{
	protected static Scanner scan = new Scanner(System.in);

	protected static int safeInteger(){
		while (true){
			try {
				return Integer.valueOf(scan.nextLine());
			}
			catch (Exception e){
				System.out.printf("Enter the correct integer: ");
			}
		}
	}
	
	protected static String safeAlphabetString(){
		String alphabetString;
		do {
			alphabetString = scan.nextLine();
			if (alphabetString.equals("") || alphabetString.matches(".?[^A-z ]+.?")){
				System.out.printf("Enter the correct alphabet string: ");
			}
		}
		while (alphabetString.equals("") || alphabetString.matches(".?[^A-z ]+.?"));
		return alphabetString;
	}
}

File: Go/src/util/input/InputValidator.go
package input

import (
	"bufio"
	"fmt"
	"os"
	"regexp"
	"strconv"
	"strings"
)

func SafeInteger() int {
	var integerResult int
	for {
		integer, err := strconv.Atoi(InputValue())
		if err == nil {
			integerResult = integer
			break
		}
		fmt.Printf("Enter the correct integer: ")
	}
	return integerResult
}

func SafeAlphabetString() string {
	var regex, err = regexp.Compile(".?[^A-z ]+.?")
	var alphabetString string = ""
	if err == nil {
		for isOk := true; isOk; isOk = (strings.EqualFold(alphabetString, "") || regex.MatchString(alphabetString)) {
			alphabetString = InputValue()
			if strings.EqualFold(alphabetString, "") || regex.MatchString(alphabetString) {
				fmt.Printf("Enter the correct alphabet string: ")
			}
		}
	}
	return alphabetString
}

func InputValue() string {
	var inputResult string
	scanner := bufio.NewScanner(os.Stdin)
	if scanner.Scan() {
		inputResult = scanner.Text()
	}
	return inputResult
}
